### PR TITLE
Employee CRUD form functionality completed with validations and tested responses from API

### PR DESCRIPTION
**Pull request by senseihimanshu**
Please ignore admin.component.html, admin.component.scss, admin.component.ts. These were just for testing purposes and will get removed in upcoming commits.

### CODE REVIEW

File: api/controllers/base.js [hunk 1]
```diff
@@ -1,95 +1,175 @@
...
+  }
 
-    async get(req, res){
-        await this.model.get(req.params.id);
+  async get(req, res) {
+    try {
+      const data = await this.model.get(req.params.id);
+      return res.send({
+        success: true,
+        payload: {
```
Comment (sonalichawla16) [inline api/controllers/base.js:33]: Add Employee to each message payload to make consistency throughout application

File: web/src/app/utilities/http-interceptor.service.ts [hunk 2]
```diff
@@ -40,11 +49,8 @@ export class HttpInterceptorService implements HttpInterceptor {
         return event;
       }),
       catchError((error: HttpErrorResponse) => {
-        if(error.status == 401){
-          console.log("Unauthorized");
-        }
-        else if(error.status == 404){
-          console.log("Not Found");
+        if (error.status == 401) {
+        } else if (error.status == 404) {
```
Comment (sonalichawla16) [inline web/src/app/utilities/http-interceptor.service.ts:53]: Blank If block ??

Comment (senseihimanshu) [reply web/src/app/utilities/http-interceptor.service.ts:53]: This file is not written by me. I will ask the team. I just removed console from it!

File: api/controllers/base.js (hunk 1 above)
Comment (senseihimanshu) [reply api/controllers/base.js:33]: Ok, Sure! Changes will be reflected in future commits.

File: api/controllers/interview.js [hunk 3]
```diff
@@ -1,110 +1,94 @@
...
+        let updatedInterviewStatus = await interviewModel.updateOne(
+          { _id: req.params.id },
+          updateObj
+        );
+        res.send({
+          success: true,
+          payload: {
+            body: updatedInterviewStatus,
+            message: "updated interview successfully",
+          },
```
Comment (sonalichawla16) [inline api/controllers/interview.js:40]: Take care of casing please

File: api/controllers/interview.js [hunk 4]
```diff
@@ -1,110 +1,94 @@
...
+      } else {
+        let deletedInterviewStatus = await interviewModel.deleteOne({
+          _id: req.params.id,
+        });
+        res.send({
+          success: true,
+          payload: {
+            body: deletedInterviewStatus,
+            message: "deleted interview successfully",
+          },
```
Comment (sonalichawla16) [inline api/controllers/interview.js:65]: Here also

File: api/controllers/interview.js [hunk 5]
```diff
@@ -1,110 +1,94 @@
...
+        });
+      } else {
+        res.send({
+          success: true,
+          payload: {
+            body: interview,
+            message: "showing interview details",
+          },
+        });
```
Comment (sonalichawla16) [inline api/controllers/interview.js:88]: same thing is repeated here aswell

File: web/src/app/utilities/http-interceptor.service.ts (hunk 2 above)
Comment (sonalichawla16) [reply web/src/app/utilities/http-interceptor.service.ts:53]: okay.
Confirm it and get it rectified

File: api/controllers/interview.js (hunk 3 above)
Comment (senseihimanshu) [reply api/controllers/interview.js:40]: Ok, Sure! I will update these give me some time.

File: web/src/app/utilities/http-interceptor.service.ts (hunk 2 above)
Comment (senseihimanshu) [reply web/src/app/utilities/http-interceptor.service.ts:53]: We need to add the code for redirection after login. So, We will add the code of it in upcoming commits.

**Pull request merged**